### PR TITLE
feat(pdf): working on export to pdf action

### PR DESCRIPTION
**Pull request by nestoragent**
# Contribution checklist

## Brief description

feat(pdf): working on export to pdf action

## Checklist

* [ ] - UI test added to added/changed functionality?
* [ ] - Screenshot is added to PR if applicable ?
* [ ] - Does a local formatting is applied (rush format)
* [ ] - Does a local svelte-check is applied (rush svelte-check)
* [ ] - Does a local UI tests are executed [UI Testing](../tests/readme.md)
* [ ] - Does the code work? Check whether function and logic are correct.
* [ ] - Does Changelog.md is updated with changes?
* [ ] - Does the translations are up to date?
* [ ] - Does it well tested?
* [ ] - Tested for Chrome.
* [ ] - Tested for Safari.
* [ ] - Go through the changed code looking for typos, TODOs, commented LOCs, debugging pieces of code, etc.
* [ ] - Rebase your branch onto master and upstream branch
* [ ] - Is there any redundant or duplicate code?
* [ ] - Are required links are linked to PR?
* [ ] - Does new code is well documented ?

## Related issues

A list of closed updated issues

## Contributor requirements

* [ ] - Sign-off is provided. [DCO](https://github.com/apps/dco)
* [ ] - GPG Signature is provided. [GPG](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits)

<sub><a href="https://front.hc.engineering/guest/platform?token=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJsaW5rSWQiOiI2NWZiNGJlZDA5MjkyMzllMmYxODlmMzEiLCJndWVzdCI6InRydWUiLCJlbWFpbCI6IiNndWVzdEBoYy5lbmdpbmVlcmluZyIsIndvcmtzcGFjZSI6InBsYXRmb3JtIiwicHJvZHVjdElkIjoiIn0.YGMhn6NLuMlkF4EEa1EzSgxtQ1DXUqDBCRa4JzDxKnU">Huly&reg;: <b>UBERF-6117</b></a></sub>

### CODE REVIEW

File: plugins/view-resources/src/actionImpl.ts [hunk 1]
```diff
@@ -501,6 +501,32 @@ async function getPopupAlignment (
   }
 }
 
+function ExportToPdf (
```
Comment (haiodo) [inline plugins/view-resources/src/actionImpl.ts:504]: functions should be lower case

File: plugins/view-resources/src/actionImpl.ts [hunk 2]
```diff
@@ -501,6 +501,32 @@ async function getPopupAlignment (
...
+  }
+): void {
+  const skipCheck = false
+  showPopup(
+    contact.component.DeleteConfirmationPopup,
+    {
+      object,
+      skipCheck,
+      deleteAction: async () => {
+        console.log('worked ExportToPdf')
```
Comment (haiodo) [inline plugins/view-resources/src/actionImpl.ts:518]: No need to console log in production code.

**Comment by haiodo** [comment]
Why we need this action in platform?

File: models/view/src/index.ts [hunk 3]
```diff
@@ -782,6 +782,20 @@ export function createModel (builder: Builder): void {
...
+  createAction(
+    builder,
+    {
+      action: view.actionImpl.ExportToPdf,
+      label: view.string.ExportToPdf,
+      // icon: view.icon.Delete,
+      category: view.category.General,
+      input: 'any',
+      target: core.class.Doc,
+      context: { mode: ['context', 'browser'], group: 'edit' }
```
Comment (lexiv0re) [inline models/view/src/index.ts:794]: `group` should not be `edit`. It affects the position of the action in the context menu.

**Comment by lexiv0re** [comment]
@haiodo the idea is to define this action in the platform and make it visible conditionally (only if pdf-converter service is configured for the deployment). This action then should open a preview modal with the generated pdf from where it will be possible to download it. The modal logic will also be in the platform. Everything else will not be there.
What do you think about this approach?

**Comment by haiodo** [comment]
Closing as outdated.

**Pull request closed without merging**